C7Engine/C7Engine.csproj

```diff
@@ -1,16 +1,15 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net472</TargetFramework>
-    <LangVersion>8.0</LangVersion>
+    <TargetFramework>net7.0</TargetFramework>
```

Comment:
since we're on 4.2 and .NET7 is already out of support by now, should we go ahead and target .NET8 (LTS version) and C# 10+?

Reply:
#483